Move credentials and options to json config file

atiq-cs opened the pull request:
Move credentials and options to json config file so that driver program/project can be published to public repositories i.e., github
More details in commit log.

atiq-cs commented:
@abock please let me know if you have any question. :)

atiq-cs commented:
Just saw a bunch of nullable warning popped up. Please let me know if you want me to fix them before proceeding with this. @abock 

Thanks again :)

abock commented:
Yeah, I think this one needs a bit of cleanup around nullability andj to generally embrace the most modern C# paradigms.

atiq-cs commented:
Sounds good. Let me update this and get back to you.

atiq-cs commented:
Updated the PR.

Here's build status (nullable warnings removed),

Main Project,

    $ cd ..\Goodbye.WordPress\
    $ dotnet build
    MSBuild version 17.4.0+18d5aef85 for .NET
      Determining projects to restore...
      All projects are up-to-date for restore.
      PackageVersion: 0.0.0
      Goodbye.WordPress -> D:\Code\ws\WPExport\Goodbye.WordPress\bin\Debug\net7.0\Goodbye.WordPress.dll

    Build succeeded.
        0 Warning(s)
        0 Error(s)

    Time Elapsed 00:00:01.03

Driver project,

    $ cd ..\Goodbye.WordPress.Example

    $ dotnet build
    MSBuild version 17.4.0+18d5aef85 for .NET
      Determining projects to restore...
      All projects are up-to-date for restore.
      PackageVersion: 0.0.0
      Goodbye.WordPress -> D:\Code\ws\WPExport\Goodbye.WordPress\bin\Debug\net7.0\Goodbye.WordPress.dll
      PackageVersion: 0.0.0
      Goobye.WordPress.Example -> D:\Code\ws\WPExport\Goodbye.WordPress.Example\bin\Debug\net7.0\Goobye.WordPress.Example.dll

    Build succeeded.
        0 Warning(s)
        0 Error(s)

    Time Elapsed 00:00:02.02

atiq-cs committed:
Move credentials and options to json config file so that driver/example program can be
published to public repositories i.e., github
In doing so,
- provide a list of patterns in config so that internal / confidential strings
  can be put there
- add namespace 'WPExportApp' to share it with JsonConfig class
- add unicode examples: weird char replacements

And,
- comment 'CustomMetadata' out, for actual applications

In addition,
- update framework to 7.0
- add Nullable Annotations for System.Linq
(ref, https://stackoverflow.com/questions/58670909/nullable-reference-type-information-not-exposed-from-firstordefault)

Nits
- update Copyright
- update dep nuget packages

// Initial PR Date: 11-21-2021

atiq-cs commented:
Let me know what you think about the fix! :)
